frontend/src/core/cells/cells.ts

```diff
@@ -1689,40 +1689,51 @@ export function createUntouchedCellAtom(cellId: CellId): Atom<boolean> {
 export function createTracebackInfoAtom(
   cellId: CellId,
 ): Atom<TracebackInfo[] | undefined> {
-  // We create an intermediate atom that just computes the string
-  // so it prevents downstream recomputations.
-  const tracebackStringAtom = atom<string | undefined>((get) => {
+  // Single flat atom - proper reactivity
+  return atom((get: any) => {
```

Comment:
this now runs on every notebook change. can we instead at least do keep the intermediate atom that computes `cellDataAtom` (just returning `const data = notebook.cellRuntime[cellId]`)